src/kolibri_android/main_activity/kolibri_bus.py

```diff
@@ -49,6 +50,12 @@ def is_kolibri_url(self, url):
 
         return False
 
+    def stop_services(self):
+        self._services.STOP()
+
+    def start_services(self):
+        self._services.START()
+
```

Comment:
It's possible for an activity's `onStart()` callback to run multiple times in between calls to `onStop()`, for example if the activity is _paused_ but not stopped. Looking at `ServicesPlugin` source, it appears the iceqube engine is going to be created from scratch each time the `START()` function runs: https://github.com/learningequality/kolibri/blob/bbd63a9c1f0371dc5009f59375c43aec26092610/kolibri/utils/server.py#L255-L267.

So, I think we should add a way to keep track of whether the services plugin has been stopped or started so we aren't doing it needlessly. Perhaps add our own `PausableServicesPlugin` which keeps track of that state?

Reply:
Thanks for the feedback! I was able not only to skip the START when it wasn't stopped before, but also to partially start after stopped. To prevent calling `schedule_ping()` and `schedule_vacuum()` multiple times. See the fixup commit.